providers/gitea_server.go

```diff
@@ -0,0 +1,72 @@
+package providers
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/drone/drone-go/drone"
+	"github.com/drone/go-scm/scm"
+	"github.com/drone/go-scm/scm/driver/gitea"
+	"github.com/drone/go-scm/scm/transport"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type giteaDiffs struct {
+	Diffs []struct {
+		Destination struct {
+			Name string `json:"toString"`
+		} `json:"destination"`
+	} `json:"diffs"`
+}
+
+func GetGiteaFilesChanged(repo drone.Repo, build drone.Build, token string, uri string) ([]string, error) {
+	var err error
+
+	newctx := context.Background()
+	client, err := gitea.New(uri)
+	if err != nil {
+		return nil, err
+	}
+	client.Client = &http.Client{
+		Transport: &transport.BearerToken{
+			Token: token,
+		},
+	}
+
+	var changes []*scm.Change
+	var result *scm.Response
+
+	if build.Before == "" || build.Before == scm.EmptyCommit {
+		changes, result, err = client.Git.ListChanges(newctx, repo.Slug, build.After, scm.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		changes, result, err = client.Git.CompareChanges(newctx, repo.Slug, build.Before, build.After, scm.ListOptions{})
```

Comment:
FYI, neither of these calls are implemented in go-scm.

https://github.com/drone/go-scm/blob/be60c693c948b2746bd1c68684dcc51478757699/scm/driver/gitea/git.go#L76-L82